test/unittests/test_hive_map.py

```
@@ -0,0 +1,350 @@
+from hivemind_bus_client.message import HiveMessage, HiveMessageType
+from hivemind_core.hive_map import HiveMapper, NodeInfo
+from hivemind_core.protocol import HiveMindListenerProtocol
+
+
+# ---------------------------------------------------------------------------
+# Helpers
+# ---------------------------------------------------------------------------
+
+def _make_ping(flood_id: str, peer: str, site_id: str = "",
+               timestamp: float = 1000.0,
+               route: list = None) -> HiveMessage:
```

Ruff (0.15.6)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)